content/blog/2023/03/camel-observation/index.md

```diff
@@ -0,0 +1,201 @@
+example of such a key value (for example, `/user/user1234`, `/user/user2345`, and so on). *Low cardinality* means that a key value will  have a bounded number of possible values. A *templated* HTTP URL (such as `/user/{userId}`) is a good example of such a key value.
+
+Assuming that the code above does instrumentation you haven't seen any tracing or metrics related setup so far. That's because we separate instrumentation from its configuration. The fact of doing metrics, tracing or logging should be a configuration problem. Let's look now how we could configure the `ObservationRegistry` to do metrics and tracing.
+
+### Setting up the ObservationRegistry
+
+The following snippet of code shows how to set up basic metrics and tracing capabilities for an `ObservationRegistry`.
+
+```java
+// Micrometer Core JAR
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+// Micrometer Observation JAR
+import io.micrometer.core.instrument.observation.DefaultMeterObservationHandler;
+import io.micrometer.observation.ObservationRegistry;
+// Micrometer Tracing JAR
+import io.micrometer.tracing.Tracer;
+
+// [Tracing]
+// Skipped the whole setup of a Micrometer Tracing tracer
+// you can check the https://micrometer.io/docs/tracing docs
+// for examples of Brave and OpenTelemetry Tracer setup
+io.micrometer.tracing.Tracer tracer = ...; // <1>
+
+// [Metrics]
+MeterRegistry meterRegistry = new SimpleMeterRegistry(); // <2>
+
+// Setting up ObservationRegistry
+ObservationRegistry registry = ObservationRegistry.create(); // <3>
+// Adding metrics support
+registry.observationConfig().observationHandler(new DefaultMeterObservationHandler(meterRegistry)); // <4>
+// Adding basic tracing support
+registry.observationConfig().observationHandler(new DefaultTracingObservationHandler(tracer)); // <5>
+```
+
+- <1> - for tracing a `Tracer` is needed. Check the [docs](https://micrometer.io/docs/tracing) on more information on how to set it up
+- <2> - for metrics a `MeterRegistry` is needed. Check the [docs](https://micrometer.io/docs/concepts) on more information on how to wotk with Micrometer
+- <3> - an `ObservationRegistry` is required to create `Observation`. Check the [docs](https://micrometer.io/docs/observation) on more information on how to set it up
+- <4> - to add capabilities to `ObservationRegistry` one needs to add handlers. Here you have an example of a handler for metrics
+- <5> - Here you have an example of a handler for tracing. This is a basic sample, for moew information check this [part of the docs](https://micrometer.io/docs/tracing#_handler_configuration)
+
+
+## How does Micrometer Observation work with Camel?
+
+Apache Camel comes with an abstraction of a [`Tracer`](https://camel.apache.org/components/3.20.x/others/tracing.html). It automatically creates spans for Camel Routes.
```

Comment:
We don't allow absolute links to https://camel.apache.org. This is so that the link checker can check these links via filesystem.

```suggestion
Apache Camel comes with an abstraction of a [`Tracer`](/components/3.20.x/others/tracing.html). It automatically creates spans for Camel Routes.
```